tests/context.test.tsx

```diff
@@ -152,3 +153,19 @@ it('throws error when not using provider', async () => {
   )
   await findByText('errored')
 })
+
+it('useCallback with useStore infers types correctly', async () => {
+  const { useStore } = createContext<CounterState>()
+  function _Counter() {
+    let x = useStore(useCallback((state) => state.count, []))
+    expectAreTypesEqual<typeof x, number>().toBe(true)
```

Comment:
We have to use this because writing `x: number = useStore(..)` won't test the inference as even when the inference fails `x` will be inferred as `any` and it'll still compile.

Reply:
Interesting. Nice to learn a new technique.

Just wondered why it was inferred as `any` instead of `unknown` before this PR...

Reply:
> Just wondered why it was inferred as any instead of unknown before this PR...

Oops sorry I missed this. It's because in `useCallback`'s callback's constraint is `(...a: any[]) => any` as you can see [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/548b1d876886a4cc4340e33cf6ca3991782bc0d7/types/react/v17/index.d.ts#L1105). Hence [here](https://tsplay.dev/mAyQvm) you will see `x` is `any` in `useFoov37`. But the correct constraint is for an unknown function is `(...a: never[]) => unknown`, in that case the code [wouldn't even compile](https://tsplay.dev/mp9LgW). (I hope this is what you were asking)